File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/provider/ContentProviderModel.java
package com.raizlabs.android.dbflow.test.provider;

import android.net.Uri;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.annotation.provider.ContentUri;
import com.raizlabs.android.dbflow.annotation.provider.TableEndpoint;
import com.raizlabs.android.dbflow.structure.provider.BaseProviderModel;
import com.raizlabs.android.dbflow.structure.provider.ContentUtils;

/**
 * Description:
 */
@TableEndpoint(name = ContentProviderModel.NAME, contentProvider = ContentDatabase.class)
@Table(database = ContentDatabase.class, name = ContentProviderModel.NAME)
public class ContentProviderModel extends BaseProviderModel<ContentProviderModel> {

    public static final String NAME = "ContentProviderModel";

    @ContentUri(path = NAME, type = ContentUri.ContentType.VND_MULTIPLE + NAME)
    public static final Uri CONTENT_URI = ContentUtils.buildUriWithAuthority(ContentDatabase.AUTHORITY);

    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    String notes;

    @Column
    String title;

    @Override
    public Uri getDeleteUri() {
        return TestContentProvider.ContentProviderModel.CONTENT_URI;
    }

    @Override
    public Uri getInsertUri() {
        return TestContentProvider.ContentProviderModel.CONTENT_URI;
    }

    @Override
    public Uri getUpdateUri() {
        return TestContentProvider.ContentProviderModel.CONTENT_URI;
    }

    @Override
    public Uri getQueryUri() {
        return TestContentProvider.ContentProviderModel.CONTENT_URI;
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/provider/ContentProviderModel.kt
package com.raizlabs.android.dbflow.provider

import android.net.Uri

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.annotation.provider.ContentUri
import com.raizlabs.android.dbflow.annotation.provider.TableEndpoint
import com.raizlabs.android.dbflow.structure.provider.BaseProviderModel
import com.raizlabs.android.dbflow.structure.provider.ContentUtils

/**
 * Description:
 */
@TableEndpoint(name = ContentProviderModel.NAME, contentProvider = ContentDatabase::class)
@Table(database = ContentDatabase::class, name = ContentProviderModel.NAME)
class ContentProviderModel : BaseProviderModel() {

    @Column
    @PrimaryKey(autoincrement = true)
    var id: Long = 0

    @Column
    var notes: String? = null

    @Column
    var title: String? = null

    override fun getDeleteUri(): Uri {
        return TestContentProvider.ContentProviderModel.CONTENT_URI
    }

    override fun getInsertUri(): Uri {
        return TestContentProvider.ContentProviderModel.CONTENT_URI
    }

    override fun getUpdateUri(): Uri {
        return TestContentProvider.ContentProviderModel.CONTENT_URI
    }

    override fun getQueryUri(): Uri {
        return TestContentProvider.ContentProviderModel.CONTENT_URI
    }

    companion object {

        const val NAME = "ContentProviderModel"

        @ContentUri(path = NAME, type = ContentUri.ContentType.VND_MULTIPLE + NAME)
        val CONTENT_URI = ContentUtils.buildUriWithAuthority(ContentDatabase.AUTHORITY)
    }
}
